### DISCUSSION TITLE
Is posible to override BaseHTTPMiddleware class

### DISCUSSION THREAD

**Discussion by bozzlab** [fastapi/fastapi#7774] [Questions]
from my issue https://github.com/tiangolo/fastapi/issues/954. I found that middleware decarator are required ( Request and call_next ) on the function. Which call_next() will return a Response value by pass a Request in function. But the call_next(), It's return StreamingResponse ( from Response Class). There are no attribute for Body or any instead to get body object from StreamingResponse. 
So if i would like to override BaseHTTPMiddleware, or it's have way to get body from StreamingResponse?

**Comment by bozzlab** [comment]
I found this https://www.starlette.io/middleware/#basehttpmiddleware. But the dispatch() are require call_next() too. Oh...

**Comment by tiangolo** [comment]
I understand you solved your problem :tada:

Thanks for reporting back and closing the issue.
